jams/core.py

```diff
@@ -557,139 +531,30 @@ def dense(self, value):
         '''Setter for dense'''
         self.__dense = value
 
-    @classmethod
-    def fields(cls):
-        '''Fields of a JamsFrame: (time, duration, value, confidence)
-
-        Returns
-        -------
-        fields : list
-            The only permissible fields for a JamsFrame:
-            `time`, `duration`, `value`, and `confidence`
-        '''
-        return ['time', 'duration', 'value', 'confidence']
-
-    @classmethod
-    def from_dict(cls, *args, **kwargs):
-        '''Construct a new JamsFrame from a dictionary or list of dictionaries.
-
-        This is analogous to pd.DataFrame.from_dict, except the returned object
-        has the type `JamsFrame`.
-
-        See Also
-        --------
-        pandas.DataFrame.from_dict
-        from_dataframe
-        '''
-        new_frame = super(JamsFrame, cls).from_dict(*args, **kwargs)
-
-        return cls.from_dataframe(new_frame)
-
-    @classmethod
-    def from_dataframe(cls, frame):
-        '''Convert a pandas DataFrame into a JamsFrame.
-
-        Note: this operation is destructive, in that the input
-        DataFrame will have its type and data altered.
-
-        Parameters
-        ----------
-        frame : pandas.DataFrame
-            The input DataFrame.  Must have the appropriate JamsFrame fields:
-            'time', 'duration', 'value', and 'confidence'.
-
-            'time' and 'duration' fields should be of type `float` and measured
-            in seconds.
-
-        Returns
-        -------
-        jams_frame : JamsFrame
-            The input `frame` modified to form a JamsFrame.
-
-        See Also
-        --------
-        from_dict
-        '''
-        # Encode time properly
-        frame.time = pd.to_timedelta(frame.time, unit='s')
-        frame.duration = pd.to_timedelta(frame.duration, unit='s')
-
-        # Properly order the columns
-        frame = frame[cls.fields()]
+    def add_observation(self, time=None, duration=None, value=None,
+                        confidence=None):
+        idx = self.obs.bisect_key(time)
```

Comment:
if nothing else, I think we can use `SortedListWithKey` directly ([docs](http://www.grantjenks.com/docs/sortedcontainers/sortedlistwithkey.html#sortedcontainers.SortedListWithKey.L.add)). I poked around, and calling `add` sorts items in-line, so `self.obs.add(Observation(...))
` should work.

additionally, i think if AnnotationData is a sublcass of `SortedListWithKey`, then we might subclass `add` directly? and maybe call it `ObservationArray`? idk?